templates/deployments/generate.sh.tpl

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+# This script generates kubernetes config files for all jsonnet files in the deployments folder.
+# This way we can easily check what changes are being made to which environment.
+
+ROOT=./deployments/{{ .Config.Name }}
+OUTPUT=$ROOT/kubecfg
+DOWNLOAD_LIBS=1
+
+if [ $# -eq 1 ] && [ "$1" = "--quick" ]; then
+  DOWNLOAD_LIBS=0
+fi
+
+# Exit when any command fails
+set -e
+
+# Prepare output folder
+rm -rf $OUTPUT
+mkdir -p $OUTPUT
+
+generate() {
+  line=$1
+
+  channel=$(echo "$line" | jq -r '.channel')
+  cluster=$(echo "$line" | jq -r '.cluster')
+  environment=$(echo "$line" | jq -r '.environment')
+  name=$(echo "$line" | jq -r '.name')
+  region=$(echo "$line" | jq -r '.region')
+  dev_email=$(echo "$line" | jq -r '.dev_email')
+
+  echo "Generating kubernetes config for '$name'"
+  kubecfg -J ./deployments/libs show \
+    -V "appImageRegistry=gcr.io/outreach-docker" -V "bento=$name" -V "channel=$channel" -V "cluster=$cluster" \
+    -V "environment=$environment" -V "namespace={{ .Config.Name }}--$name" -V "region=$region" -V "ts=1656492859" -V "version=v0.0.1" \
+    -V "dev_email=$dev_email" \
+    $ROOT/{{ .Config.Name }}.jsonnet -oyaml >"$OUTPUT/$name.yaml"
+}
+
+# Download libsonnet files
+if [ $DOWNLOAD_LIBS -eq 1 ]; then
+  rm -rf ./deployments/libs
+  mkdir -p ./deployments/libs/kubernetes
+
+  outreach_libs=(app.libsonnet outreach.libsonnet kube.libsonnet cluster.libsonnet database.libsonnet)
+  for lib in "${outreach_libs[@]}"; do
+    curl -s -o "./deployments/libs/kubernetes/$lib" https://raw.githubusercontent.com/getoutreach/jsonnet-libs/master/kubernetes/"$lib"
+  done
+  curl -s -o "./deployments/libs/kubernetes/clusters.yaml" http://k8s-clusters.outreach.cloud/
+  curl -s -o "./deployments/libs/kubecfg.libsonnet" https://raw.githubusercontent.com/kubecfg/kubecfg/master/lib/kubecfg.libsonnet
+
+  find ./deployments/libs -name '*.libsonnet' -exec jsonnetfmt -i {} +
+fi
+
+generate '{"name":"bento1a", "cluster":"none", "region":"none", "channel":"white", "environment":"development", "dev_email":"stub@email"}'
```

Comment:
You may want to use disco to retrieve the canonical list of bentos for future proofing. Just a suggestion

Reply:
Thanks, I don't have time for that right now, but I might come back to that later.